pkg/testcase/certrotate.go

```diff
@@ -0,0 +1,87 @@
+package testcase
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rancher/distros-test-framework/factory"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestCertRotate() {
+	cluster := factory.AddCluster(GinkgoT())
+	serverIPs := cluster.ServerIPs
+	agentIPs := cluster.AgentIPs
+	product, err := shared.GetProductObject()
+	Expect(err).NotTo(HaveOccurred(), "error getting product from config")
+
+	certRotate(product, serverIPs)
+
+	ip, manageError := product.ManageService("restart", "agent", agentIPs)
+	Expect(manageError).NotTo(HaveOccurred(), fmt.Sprintf("error restarting agent node ip %s", ip))
+
+	verifyTLSDirContent(product, serverIPs)
+}
+
+// certRotate Rotate certificate for etcd only and cp only nodes
+func certRotate(product shared.Product, ips []string) {
+	ip, stopError := product.ManageService("stop", "server", ips)
+	Expect(stopError).NotTo(HaveOccurred(), fmt.Sprintf("error stopping %s service for node ip: %s", product, ip))
+
+	ip, rotateError := product.CertRotate(ips)
+	Expect(rotateError).NotTo(HaveOccurred(), fmt.Sprintf("error running certificate rotate for %s service on %s", product, ip))
+
+	ip, startError := product.ManageService("start", "server", ips)
+	Expect(startError).NotTo(HaveOccurred(), fmt.Sprintf("error starting %s service for node ip: %s", product, ip))
+}
+
+// verifyIdenticalFiles Verify the actual and expected identical file lists match
+func verifyIdenticalFiles(identicalFileList string) {
+	expectedFileList := []string{
+		"client-ca.crt", "client-ca.key", "client-ca.nochain.crt",
+		"client-supervisor.crt", "client-supervisor.key",
+		"dynamic-cert.json",
+		"peer-ca.crt", "peer-ca.key",
+		"server-ca.crt", "server-ca.key",
+		"request-header-ca.crt", "request-header-ca.key",
+		"server-ca.crt", "server-ca.key", "server-ca.nochain.crt",
+		"service.current.key", "service.key"}
+
+	identicalFileList = strings.TrimSpace(identicalFileList)
```

Comment:
maybe something like , to give a better name rather then too generic "slice" 
```
	newFileList := strings.Split(strings.TrimSpace(identicalFileList), "\n")

	shared.VerifyFileMatchWithPath(newFileList[1:], expectedFileList, "")
```

Reply:
fixed